scripts/test-fallback-scsv.py

```diff
@@ -69,23 +75,42 @@ def main():
 
     conversations = {}
 
-    ext = {ExtensionType.renegotiation_info: None}
     conversation = Connect(host, port)
     node = conversation
-    ciphers = [CipherSuite.TLS_RSA_WITH_AES_128_CBC_SHA,
-               CipherSuite.TLS_EMPTY_RENEGOTIATION_INFO_SCSV,
-               CipherSuite.TLS_AES_128_GCM_SHA256]
-    node = node.add_child(ClientHelloGenerator(ciphers, version=(3, 3)))
+    if dhe:
+        ext = {}
+        groups = [GroupName.secp256r1,
+                  GroupName.ffdhe2048]
+        ext[ExtensionType.supported_groups] = SupportedGroupsExtension()\
+            .create(groups)
+        ext[ExtensionType.signature_algorithms] = \
+            SignatureAlgorithmsExtension().create(RSA_SIG_ALL)
+        ext[ExtensionType.signature_algorithms_cert] = \
+            SignatureAlgorithmsCertExtension().create(RSA_SIG_ALL)
+        ciphers = [CipherSuite.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,
+                   CipherSuite.TLS_DHE_RSA_WITH_AES_128_CBC_SHA,
+                   CipherSuite.TLS_EMPTY_RENEGOTIATION_INFO_SCSV,
+                   CipherSuite.TLS_AES_128_GCM_SHA256]
+    else:
+        ext = None
+        ciphers = [CipherSuite.TLS_RSA_WITH_AES_128_CBC_SHA,
+                   CipherSuite.TLS_EMPTY_RENEGOTIATION_INFO_SCSV,
+                   CipherSuite.TLS_AES_128_GCM_SHA256]
```

Comment:
This looks like a TLS1.3 ciphersuite but no TLS1.3 seems to be enabled. Is its addition here intentional?

Reply:
yes, it's to check that the server does not use presence of TLS 1.3 ciphersuites as a proxy for supported_versions with TLS 1.3